src/components/survey/owner.tsx

```diff
@@ -1,13 +1,192 @@
-import * as React from "react";
-import { Typography } from "@mui/material";
+import { Stack, TextField, Typography, useMediaQuery } from "@mui/material";
+import Api from "api";
+import { useAppSelector } from "app/hooks";
+import { ErrorContext } from "components/error-handler/error-handler";
+import WithDebounce from "components/generic/with-debounce";
+import { selectKeycloak } from "features/auth-slice";
+import { OwnerInformation } from "generated/client";
 import strings from "localization/strings";
+import * as React from "react";
+import theme from "theme";
+
+/**
+ * Component properties
+ */
+interface Props {
+  surveyId: string;
+}
 
 /**
  * Component for owner information
  */
-const Owner: React.FC = () => {
+const Owner: React.FC<Props> = ({ surveyId }) => {
+  const keycloak = useAppSelector(selectKeycloak);
+  const errorContext = React.useContext(ErrorContext);
+  const [ ownerInformation, setOwnerInformation ] = React.useState<OwnerInformation | undefined>(undefined);
+
+  /**
+   * Create new owner information
+   */
+  const fetchOwnerInformation = async () => {
+    if (!keycloak?.token || !surveyId) {
+      return;
+    }
+
+    try {
+      const fetchedOwnerInformationArray = await Api.getOwnersApi(keycloak.token).listOwnerInformation({
+        surveyId: surveyId
+      });
+      setOwnerInformation(fetchedOwnerInformationArray[0]);
+    } catch (error) {
+      errorContext.setError(strings.errorHandling.owners.create, error);
+    }
+  };
+
+  React.useEffect(() => {
+    fetchOwnerInformation();
+  }, []);
+
+  /**
+   * Event Handler set survey prop
+   */
+  const onOwnerInfoPropChange: React.ChangeEventHandler<HTMLTextAreaElement | HTMLInputElement> = ({ target }) => {
+    const { value, name } = target;
+
+    if (!keycloak?.token || !ownerInformation?.id) {
+      return;
+    }
+
+    const updatedOwnerInformation: OwnerInformation = { ...ownerInformation, [name]: value };
+    setOwnerInformation(updatedOwnerInformation);
+
+    try {
+      Api.getOwnersApi(keycloak.token).updateOwnerInformation({
+        surveyId: surveyId,
+        ownerId: ownerInformation.id,
+        ownerInformation: updatedOwnerInformation
+      });
+    } catch (error) {
+      errorContext.setError(strings.errorHandling.owners.update, error);
+    }
```

Comment:
Move into own function and use it from both onOwnerInfoPropChange and onOwnerInfoContactPersonPropChange